integrations/access/opsgenie/app.go

```diff
@@ -75,6 +80,23 @@ func NewOpsgenieApp(ctx context.Context, conf *Config) (*App, error) {
 		PluginName: pluginName,
 		conf:       *conf,
 	}
+	teleClient, err := conf.GetTeleportClient(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	opsgenieApp.accessMonitoringRules = common.NewAccessMonitoringRuleHandler(common.AccessMonitoringRuleHandlerConfig{
+		Client:              teleClient,
+		PluginType:          string(conf.BaseConfig.PluginType),
+		RuleAppliesCallback: opsgenieApp.amrAppliesToThisPlugin,
+		FetchRecipientCallback: func(_ context.Context, name string) (*common.Recipient, error) {
```

Comment:
This code looks like a copy of the `opsgenie.Bot.FetchRecipient()`.

I looked at the bot/app situation right now and understand that you don't have a bot there and it's terribly complex to get one, but can you extract a common function for the bot and the app to build a recipient by schedule to avoid duplication?